StantUiIosLibrary/Classes/Tables/ConstructionTableViewWithCollapsedSearchBar/ConstructionTableViewWithCollapsedSearchBar.swift

```diff
@@ -85,28 +88,47 @@ public class ConstructionTableViewWithCollapsedSearchBar: UIView {
                              trailing: self.trailingAnchor,
                              padding: UIEdgeInsets(top: DefaultSearchBar.searchViewHeight, left: 0, bottom: 0, right: 0))
         }
+
     }
     
     fileprivate func configureEmptyMessageLabel() {
+        emptyMessageLabel?.removeFromSuperview()
         emptyMessageLabel = UILabel(frame: CGRect(x: 0,
                                                   y: DefaultSearchBar.searchViewHeight,
                                                   width: self.frame.width,
                                                   height: self.frame.height))
         
-        guard let emptyMessageLabel = emptyMessageLabel, let tableView = tableView else { return }
+        guard let emptyMessageLabel = emptyMessageLabel else { return }
         emptyMessageLabel.configure(text: emptyMessage,
                                     alignment: .left,
                                     size: 16,
                                     weight: .regular,
                                     color: UIColor.darkStant)
         self.addSubview(emptyMessageLabel)
-        emptyMessageLabel.anchor(top: tableView.topAnchor,
-                                 leading: self.leadingAnchor,
-                                 trailing: self.trailingAnchor,
-                                 padding: UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16),
-                                 size: CGSize(width: 0, height: 50))
         emptyMessageLabel.numberOfLines = 0
-        emptyMessageLabel.isHidden = true
+        self.setEmptyMessageLabelVisibility()
+    }
+    
+    public func updateConstructionSiteList(_ constructionSites: [Construction]) {
+        self.constructionSiteList                    = constructionSites
+        self.tableView?.filteredConstructionSiteList = constructionSites
+
+        self.tableView?.reloadData()
+        self.setEmptyMessageLabelVisibility()
+    }
+    
+    func setEmptyMessageLabelVisibility() {
```

Comment:
Maybe you can use the guard concept for the first if you did to avoid if/else and make the code more easy to read. 

```suggestion
func setEmptyMessageLabelVisibility() {
  if tableView?.filteredConstructionSiteList.count > 0 ?? true {
    emptyMessageLabel?.isHidden = true
    return 
  }
  
  guard let emptyMessageLabel = emptyMessageLabel, let tableView = tableView else { return }
  emptyMessageLabel.anchor(top:      tableView.topAnchor,
                           leading:  self.leadingAnchor,
                           trailing: self.trailingAnchor,
                           padding:  UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16),
                           size:     CGSize(width: 0, height: 50))

    emptyMessageLabel.isHidden = false
}
```

Reply:
Nice suggestion, thanks.